python_tests/test_api.py

```diff
@@ -261,6 +263,61 @@ def test_save_trial_user_attrs_empty(self) -> None:
         self.assertEqual(status, 400)
         assert study.trials[0].user_attrs == {}
 
+    def _test_save_trial_note(
```

Comment:
A nit, but could we consider renaming this method as follows? It isn't a test method but it's more like a utility function.

```suggestion
    def _save_trial_note(
```